### PR TITLE
CATROID-264 Fixed rename project with special characters

### PR DESCRIPTION
**Pull request by swlz**
added parameterized tests

### CODE REVIEW

**Comment by swlz** [comment]
One failing test but this one also fails on [develop](https://jenkins.catrob.at/job/Catroid/job/develop/892/testReport/)

**Comment by DinosaurierRex** [comment]
I think rebasing should do the trick. Fixes for these tests already merged since yesterday

**Comment by swlz** [comment]
ah, thank you. :)

**Comment by DinosaurierRex** [comment]
sorry for the misinformation but i recreated the bug yesterday and fixed it locally. Theres an timing issue and it seems that my syncTask won't work on Jenkins

File: catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectRenameTaskTest.java [hunk 1]
```diff
@@ -39,53 +40,40 @@
...
 	public void projectRenameTaskTest() throws IOException {
 		File renamedDirectory = ProjectRenameTask.task(defaultProject.getDirectory(), renamedProjectName);
 		assertEquals(renamedProjectName, renamedDirectory.getName());
-	}
-
-	@Test
-	public void projectRenameSlashTaskTest() throws IOException {
-		File renamedDirectory = ProjectRenameTask.task(defaultProject.getDirectory(), slashRenamedProject);
-		assertEquals(slashEncodedRenamedProject, renamedDirectory.getName());
+		assertTrue(ProjectLoadTask.task(renamedDirectory, InstrumentationRegistry.getTargetContext()));
```
Comment (DinosaurierRex) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectRenameTaskTest.java]: Thanks :)

File: catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectRenameSpecialCharactersTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class ProjectRenameSpecialCharactersTaskTest {
```
Comment (DinosaurierRex) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/ProjectRenameSpecialCharactersTaskTest.java]: Like that one :+1:

File: catroid/src/main/java/org/catrobat/catroid/io/XstreamSerializer.java [hunk 3]
```diff
@@ -585,15 +586,24 @@ public static boolean renameProject(File xmlFile, String dstName) throws IOExcep
...
+		StringFinder stringFinder = new StringFinder();
+
+		if (!stringFinder.findBetween(currentXml, PROGRAM_NAME_START_TAG, PROGRAM_NAME_END_TAG)) {
+			return false;
+		}
+
+		String srcName = stringFinder.getResult();
+		dstName = new XStream().toXML(dstName);
+		dstName = dstName.replace("<string>", "");
+		dstName = dstName.replace("</string>", "");
```
Comment (DinosaurierRex) [inline catroid/src/main/java/org/catrobat/catroid/io/XstreamSerializer.java]: May extract your string operations in an extra function, just to make it a bit more clean to read

**Comment by wslany** [comment]
Thanks!!

**Pull request merged**